spring-cloud-function-samples/function-sample-gcp/pom.xml

```diff
@@ -22,49 +22,30 @@
 
 	<build>
 		<plugins>
+			<plugin>
+				<groupId>org.springframework.boot</groupId>
+				<artifactId>spring-boot-maven-plugin</artifactId>
+				<configuration>
+					<outputDirectory>target/deploy</outputDirectory>
+				</configuration>
+				<dependencies>
+					<dependency>
+						<groupId>org.springframework.cloud</groupId>
+						<artifactId>spring-cloud-function-adapter-gcp</artifactId>
+						<version>${project.version}</version>
+					</dependency>
+				</dependencies>
+			</plugin>
+
 			<plugin>
 				<groupId>com.google.cloud.functions</groupId>
 				<artifactId>function-maven-plugin</artifactId>
 				<version>0.9.1</version>
 				<configuration>
-					<functionTarget>org.springframework.cloud.function.adapter.gcp.FunctionInvoker</functionTarget>
+					<functionTarget>org.springframework.cloud.function.adapter.gcp.GcfJarLauncher</functionTarget>
```

Comment:
Is the function plugin actually starting the jar?

Reply:
No I believe `mvn function:run` is equivalent to deploying from Source, so it doesn't produce a jar. However this still works because the GcfLauncher is still a legitimate entry point to the function (i.e. it has a main method).